src/OrleansAzureUtils/Providers/Lease/AzureBlobLeaseProvider.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    public class AzureBlobLeaseProviderConfig
+    {
+        public string DataConnectionString { get; set; }
+        public string BlobContainerName { get; set; }
+    }
+    public class AzureBlobLeaseProvider : ILeaseProvider
+    {
+        private CloudBlobContainer container;
+        private AzureBlobLeaseProviderConfig providerConfig;
+        private CloudBlobClient blobClient;
+        public AzureBlobLeaseProvider(AzureBlobLeaseProviderConfig config)
+        {
+            var account = CloudStorageAccount.Parse(config.DataConnectionString);
+            this.blobClient = account.CreateCloudBlobClient();
+            this.providerConfig = config;
+        }
+
+        private async Task InitContainerIfNotExistsAsync()
+        {
+            if (this.container == null)
+            {
+                this.container = blobClient.GetContainerReference(this.providerConfig.BlobContainerName);
+                await this.container.CreateIfNotExistsAsync().ConfigureAwait(false); 
+            }
+        }
+
+        public async Task<AcquireLeaseResult[]> Acquire(string category, LeaseRequest[] leaseRequests)
+        {
+            await InitContainerIfNotExistsAsync();
+            var tasks = new List<Task<AcquireLeaseResult>>();
+            foreach (var leaseRequest in leaseRequests)
+            {
+                tasks.Add(Acquire(category, leaseRequest));
+            }
+            //Task.WhenAll will return results for each task in an array, in the same order of supplied tasks
+            return await Task.WhenAll(tasks);
+        }
+
+        private string GetBlobName(string category, string resourceKey)
+        {
+            return $"{category.ToLower()}-{resourceKey.ToLower()}.json";
+        }
+
+        private async Task<AcquireLeaseResult> Acquire(string category, LeaseRequest leaseRequest)
+        {
+            try
+            {
+                var blob = this.container.GetBlockBlobReference(GetBlobName(category, leaseRequest.ResourceKey));
+                blob.Properties.ContentType = "application/json";
```

Comment:
Should this call be something like ~CreateIfNotExist? It seems strange/erroneous to be always recreating the blob